tests/specs/geoQuery.spec.js

```diff
@@ -1376,5 +1376,29 @@ describe("GeoQuery Tests:", function() {
         cl.x("p5");
       }).catch(failTestOnCaughtError);
     });
+
+    it("Calling cancel() in the middle of firing \"key_entered\" events is allowed", function(done) {
+      var cl = new Checklist(["p1", "cancel query", "loc1 entered"], expect, done);
+
+      geoQueries.push(geoFire.query({center: [1,2], radius: 1000}));
+
+      geoFire.set({
+        "loc1": [1, 2],
+        "loc2": [1, 3],
+        "loc3": [1, 4]
+      }).then(function() {
+        cl.x("p1");
+
+        var numKeyEnteredEventsFired = 0;
+        geoQueries[0].on("key_entered", function(key, location, distance) {
+          cl.x(key + " entered");
```

Comment:
Technically I don't think we can guarantee the loc1 is the first (only if it's the only one)